### PR TITLE
add option --with-imagemagick@7 and --with-jansson for HEAD

### PR DESCRIPTION
**Pull request by shuxiao9058**
Please have a test.

### CODE REVIEW

**Comment by d12frosted** [comment]
Thank you very much! Only one test failed (and due timeout). I've restarted the job. Let's see how it goes.

File: Formula/emacs-plus.rb [hunk 1]
```diff
@@ -233,12 +238,34 @@ def install
...
       args << "--without-imagemagick"
     end
 
+    # Emacs 27.x (current HEAD) supports imagemagick7 but not Emacs 26.x
+    if build.with? "imagemagick@7"
+      imagemagick_lib_path =  Formula["imagemagick@7"].opt_lib/"pkgconfig"
+      unless build.head?
+        odie "--with-imagemagick@7 is supported only on --HEAD"
+      end
+        ohai "ImageMagick PKG_CONFIG_PATH: ", imagemagick_lib_path
```
Comment (d12frosted) [inline Formula/emacs-plus.rb]: I think something happened to indentation here.

File: Formula/emacs-plus.rb [hunk 2]
```diff
@@ -233,12 +238,34 @@ def install
...
+        ENV.prepend_path "PKG_CONFIG_PATH", imagemagick_lib_path
```
Comment (d12frosted) [inline Formula/emacs-plus.rb]: Wow, this thing looks so hacky. Never used it before :)

File: Formula/emacs-plus.rb [hunk 3]
```diff
@@ -107,11 +107,17 @@ class EmacsPlus < Formula
...
+  # Emacs 26.x does not support ImageMagick 7:
   # Reported on 2017-03-04: https://debbugs.gnu.org/cgi/bugreport.cgi?bug=25967
   depends_on "imagemagick@6" => :recommended
+  # Emacs 27.x (current HEAD) does support ImageMagick 7:
+  # Prefer ImageMagick 7 when both on
+  depends_on "imagemagick@7" => :optional
+
   depends_on "mailutils" => :optional
 
+  depends_on "jansson" => :optional
```
Comment (d12frosted) [inline Formula/emacs-plus.rb]: Once 27 is out this one should become recommended!

**Comment by d12frosted** [comment]
Yay. Tests are green. I would like to play around with it. Hopefully I will have some time on weekends.

Meanwhile, it would be really nice if you could check if it's possible to manipulate the list of recommended dependencies based on the `--HEAD` option. If it's impossible, I am OK with it.

**Comment by shuxiao9058** [comment]
the code refer to https://github.com/d12frosted/homebrew-emacs-plus/issues/117

**Comment by shuxiao9058** [comment]
> I wonder if it's possible to actually make `imagemagick@7` recommended on `HEAD` and optional on stable using ifs.

I will try later today.

**Comment by shuxiao9058** [comment]
@d12frosted I have a problem, how to show the option with imagemagick@7 and imagemagick@6?


``` ruby
  if build.head?
    # Emacs 27.x (current HEAD) does support ImageMagick 7:
    # Prefer ImageMagick 7 when both on
    depends_on "imagemagick@7" => :recommended
    depends_on "imagemagick@6" => :optional
  else
    # Emacs 26.x does not support ImageMagick 7:
    # Reported on 2017-03-04: https://debbugs.gnu.org/cgi/bugreport.cgi?bug=25967
    depends_on "imagemagick@6" => :recommended
  end
```

With the code, the option only
```
--without-imagemagick@6
	Build without imagemagick@6 support
```

**Comment by d12frosted** [comment]
Well, it makes sense. Let's not do it. Please update the README (section with options) and I am ready to merge :)

File: Formula/emacs-plus.rb (hunk 3 above)
Comment (shuxiao9058) [reply Formula/emacs-plus.rb]: > Once 27 is out this one should become recommended!

It seems that the native json support is on with default.

 https://github.com/emacs-mirror/emacs/blob/eb6bbd9fb175cacdfdc54c1187f5785ed3858f2f/configure.ac#L432

**Comment by d12frosted** [comment]
Thank you for your PR :heart: :100: It's not in master, so you can safely remove your branch.

P. S. I've actually squashed your commits and fixed conflict. Resulting commit is 368431aceec319b56c90b63ea830216d16c503a0.

**Pull request closed without merging**